ath79: add ath9k mtd-cal-data in dts for nbg6716

glefebvr1 opened the pull request:
There was no detection of the ath9k caldata from art partition in dts file
for Zyxel NBG6716. This prevented the ath9k radio from being properly
initialised and induced the following error messages:
  ath: phy1: Unable to initialize hardware; initialization status: -5
  ath9k 18100000.wmac: failed to initialize device
  ath9k: probe of 18100000.wmac failed with error -5

Signed-off-by: Guillaume Lefebvre <guillaume@zelig.ch>

adrianschmutzler commented:
ath9k setup has been done in 10-ath9k-eeprom so for: https://github.com/openwrt/openwrt/blob/master/target/linux/ath79/nand/base-files/etc/hotplug.d/firmware/10-ath9k-eeprom#L16

So, if this is wrong you should remove that. The ugly part will be to get the MAC address set to ath9k Wifi anyway ...

Commit:
ath79: fix ath9k initialization for nbg6716

ath9k initialization (caldata + mac address) was currently done as if
using ath9k-htc driver. Zyxel NBG6716 does not use it. It rather uses
the ath9k driver.
This commit fixes caldata & mac address for nbg6716 and also the
following error messages:
ath: phy1: Unable to initialize hardware; initialization status: -5
ath9k 18100000.wmac: failed to initialize device
ath9k: probe of 18100000.wmac failed with error -5

Signed-off-by: Guillaume Lefebvre <guillaume@zelig.ch>

glefebvr1 commented:
Hi, thanks for the review.
I got a bit deeper and found that ath9k on nbg6716 was initialized as if using ath9k-htc driver, which is not the case. It uses the ath9k driver, which has no firmware.
So, I changed the commit, including setting the mac address.
BTW, I noticed that the mac addresses for ethernet devices were also not set. I'll do an other PR for this.

adrianschmutzler commented on target/linux/ath79/nand/base-files/etc/hotplug.d/ieee80211/10-fix-wifi-mac:
```diff
@@ -0,0 +1,17 @@
+zyxel,nbg6716)
+	# Set mac address for ath9k device
+	[ "$PHYNBR" -eq 1 ] && \
+		macaddr_add "$(mtd_get_mac_ascii u-boot-env ethaddr)" 2 > /sys${DEVPATH}/macaddress
```
So, you use another MAC address, too? As always, I'd be happy about an overview of MAC addresses used and where they come from (flash locations, calculation offset, etc...) added to the commit message. (Something like the list here: https://openwrt.org/docs/guide-developer/mac.address#merge_your_data)

adrianschmutzler commented:
I'd still be interested what's the correct MAC address assignment, as you have different offsets in this patch and your subsequent one compared to the current state.

adrianschmutzler commented:
Since this has been stale for quite some time, I close it now.
I've sent an altered version of this together with other fixes for the device to the mailing list.

If there are some users willing to test, please take a look there.

https://patchwork.ozlabs.org/project/openwrt/list/?series=169215
